.github/workflows/release-published.yml

```diff
@@ -0,0 +1,25 @@
+name: Checks after release is published
+on:
+  release:
+    types: ['published']
+
+jobs:
+  validate-github-action:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: k14s/setup-k14s-action@v1
+        with:
+          only: vendir
+      - run: |
+          vendir version
+          version=`vendir version`
+          tag="${{ github.event.release.tag_name }}"
+          tool_version="$(echo $version | sed 's/^.*[^0-9]\([0-9]*\.[0-9]*\.[0-9]*\).*$/\1/')"
+          if [[ "v${tool_version}" == "${tag}" ]];
```

Comment:
There's an edge case where this check may not do what you expect: if you provide a minor or patch update to a previous major version (e.g. for security reasons). The setup action is designed to install the latest version by semver name by default, not the most recent version published to GitHub if that is different.

Admittedly, this is very much an edge case. However, if you want to anticipate it, I _think_ you could amend the setup step like this:

```yml
- uses: k14s/setup-k14s-action@v1
  with:
    only: vendir
    vendir: ${{ github.event.release.tag_name }}
```

Note that I didn't test this, but in theory this should install the correct version for the release tag, even if you've just published a patch for an older major/minor version.